app/controllers/planning_applications/validation/document/redactions_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module PlanningApplications
+  module Validation
+    module Document
+      class RedactionsController < AuthenticationController
+        before_action :set_planning_application
+
+        def index
+          respond_to do |format|
+            format.html
+          end
+        end
+
+        def create
+          @planning_application.assign_attributes(assign_params)
+
+          respond_to do |format|
+            format.html do
+              if @planning_application.save
+                redirect_to planning_application_validation_tasks_path(@planning_application), notice: t(".success")
+              else
+                render :index
+              end
+            end
+          end
+        end
+
+        private
+
+        def assign_params
+          documents_attributes = planning_application_params[:documents_attributes].select do |_key, value|
+            value[:file].present?
+          end
+
+          new_params = planning_application_params
+
+          new_params[:documents_attributes] = documents_attributes
+          new_params
+        end
```

Comment:
I think we can also do this like:

```ruby
# models/planning_application.rb
accepts_nested_attributes_for :documents, reject_if: proc { |attributes| attributes["file"].blank? }

# controllers/planning_applications/validation/document/redactions_controller.rb
def planning_application_params
  params.require(:planning_application)
        .permit(documents_attributes: documents_params)
end

def documents_params
  %i[file redacted publishable]
end
```

which means we don't have to do with `assign_params` stuff.
But not sure if that will mess with other ways we update documents

Reply:
ooh that's cool, haven't seen that before! i'll give it a go 😄